pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -36,9 +43,18 @@ func New(client *kubernetes.Clientset, nodeID, namespace, name, annotation strin
 // Acquire attempts to annotate the kured daemonset with lock info from instantiated DaemonSetLock using client-go
 func (dsl *DaemonSetLock) Acquire(metadata interface{}, TTL time.Duration) (acquired bool, owner string, err error) {
 	for {
-		ds, err := dsl.client.AppsV1().DaemonSets(dsl.namespace).Get(context.TODO(), dsl.name, metav1.GetOptions{})
+		var ds *v1.DaemonSet
+		var lastError error
+		err := wait.PollImmediate(k8sAPICallRetrySleep, k8sAPICallRetryTimeout, func() (bool, error) {
+			ctx, cancel := context.WithTimeout(context.Background(), k8sAPICallRetryTimeout)
+			defer cancel()
+			if ds, lastError = dsl.client.AppsV1().DaemonSets(dsl.namespace).Get(ctx, dsl.name, metav1.GetOptions{}); lastError != nil {
+				return false, nil
+			}
+			return true, nil
+		})
 		if err != nil {
-			return false, "", err
+			return false, "", fmt.Errorf("Timed out trying to get daemonset %s in namespace %s: %v", dsl.name, dsl.namespace, lastError)
```

Comment:
Use `%w` in the format string to wrap the error, rather than just stringifying it.
```suggestion
			return false, "", fmt.Errorf("timed out trying to get daemonset %s in namespace %s: %w", dsl.name, dsl.namespace, lastError)
```